File: Flutter/lib/models/stream_chat_message.dart
class StreamChatMessage {
  final String messageId;
  final String streamId;
  final String userId;
  final String userName;
  final String? userProfileImageUrl;
  final String message;
  final DateTime createdAt;

  StreamChatMessage({
    required this.messageId,
    required this.streamId,
    required this.userId,
    required this.userName,
    this.userProfileImageUrl,
    required this.message,
    required this.createdAt,
  });

  factory StreamChatMessage.fromJson(Map<String, dynamic> json) {
    // Helper to parse ID fields (handle both string GUID and int legacy formats)
    String parseId(dynamic id, {String defaultValue = ''}) {
      if (id == null) return defaultValue;
      if (id is String) return id;
      if (id is int) return id.toString(); // Legacy format
      return id.toString();
    }

    return StreamChatMessage(
      messageId: parseId(json['messageId']),
      streamId: parseId(json['streamId']),
      userId: parseId(json['userId']),
      userName: json['userName'] ?? 'Unknown User',
      userProfileImageUrl: json['userProfileImageUrl'],
      message: json['message'] ?? '',
      createdAt: DateTime.parse(
        json['createdAt'] ?? DateTime.now().toIso8601String(),
      ),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'messageId': messageId,
      'streamId': streamId,
      'userId': userId,
      'userName': userName,
      'userProfileImageUrl': userProfileImageUrl,
      'message': message,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: API/Models/StreamChatMessage.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace InstapropAPI.Models
{
    public class StreamChatMessage
    {
        [Key]
        public Guid MessageId { get; set; }

        [Required]
        public Guid StreamId { get; set; }

        [ForeignKey(nameof(StreamId))]
        public LiveStream Stream { get; set; } = null!;

        [Required]
        public Guid UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public AccountBase User { get; set; } = null!;

        [Required]
        [MaxLength(500)]
        public string Message { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

